### PR TITLE
feat: use 🛻 pickup from api

### LINKED ISSUE #2309: Use 🛻 pickup to handle pinning service requests
Review the api env, db schema, and cron jobs to find out what needs to be updated to have the api use pickup to handle pinning service requests.

## TODO

- [x] update `api/src/routes/pins-add.js` to specify ElasticIpfs as the service when calling createUpload
- [x] update `api/src/routes/pins-replace.js` to specify ElasticIpfs as the service when calling createUpload
- [x] update `cron/src/bin/pins.js` to add a cluster client configured to hit `pickup`.
- [x] update `cron/src/jobs/pins.js` to add `ElasticIpfs` to the list of Clusters to check
- [x] update `cron/src/bin/pins.js` to add a check pin statuses on pickup for via status.all
- [ ] add `staging` worker env var for `CLUSTER_API_URL` set to staging pickup url
- [ ] remove `staging` worker env var for `CLUSTER_SERVICE`
- [ ] add `prod` worker env var for `CLUSTER_API_URL` set to prod pickup url
- [ ] remove `prod` worker env var for `CLUSTER_SERVICE`

note: the env var `CLUSTER_SERVICE` takes precedence over `CLUSTER_API_URL`, so we can add a value for `CLUSTER_API_URL` whenever we like, and then delete the `CLUSTER_SERVICE` env var when we are ready for to make the switch.

Parent: https://github.com/web3-storage/pickup/issues/68

### PR DESCRIPTION
**Pull request by olizilla**
pickup pulls dags into E-IPFS, so this PR updates the api to reflect that.

- Update the pinning service add and update api routes to record `ElasticIpfs` as the service that the pin is queued on.
- Update `GET /pins/:req-id` to check for a status change from pickup and update out DB, so we can inform users early when a pin request is pinned or failed.

Cron changes are in #2339 

Fixes #2309 

TODO

- [x] add `PICKUP_BASIC_AUTH_TOKEN` as a secret to the repo for cron jobs.
- [x] add `PICKUP_BASIC_AUTH_TOKEN` as worker secret for dev, staging, and prod
- [ ] deploy
- [ ] remove CLUSTER env vars

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>

### CODE REVIEW

**Comment by olizilla** [comment]
This is blocked on pickup validator improvements https://github.com/web3-storage/pickup/issues/97

File: packages/api/src/routes/pins-add.js [hunk 1]
```diff
@@ -51,6 +51,12 @@ export async function pinsAdd(event, ctx) {
   })
 
   const upload = await db.createUpload({
+    pins: [
+      {
+        status: 'PinQueued',
+        service: 'ElasticIpfs', // via pickup
+      },
+    ],
```
Comment (olizilla) [inline packages/api/src/routes/pins-add.js:59]: We do this to override the default pin info that `createUpload` would set if we didn't provide this. New pins via pickup are provided via E-IPFS.

**Review by alanshaw — APPROVED** [review]

**Comment by olizilla** [comment]
Raised an issue for the dev worker deploy fail https://github.com/nftstorage/nft.storage/issues/2349

**Review by vasco-santos — APPROVED** [review]
LGTM ✨

**Pull request merged**